mirgecom/materials/carbon_fiber.py

```diff
@@ -29,11 +29,28 @@
 """
 
 from typing import Optional
+from abc import abstractmethod
 import numpy as np
 from meshmode.dof_array import DOFArray
+from mirgecom.wall_model import PorousWallEOS
 
 
 class Oxidation:
+    """Abstract interface for wall oxidation model.
+
+    .. automethod:: get_source_terms
+    """
+
+    @abstractmethod
+    def get_source_terms(self, temperature: DOFArray,
+            tau: DOFArray, rhoY_o2: DOFArray) -> DOFArray:  # noqa N803
+        r"""Source terms of fiber oxidation."""
+        raise NotImplementedError()
+
+
+# TODO per MTC review, can we generalize the oxidation model?
+# should we keep this in the driver?
```

Comment:
tbh, I do not see any real harm in internalizing some of our use-specific constructs.  If it makes it easier for us (and others) to use, then we should.  Is that the question, or did I miss the mark?

Reply:
The original comment was originally here: https://github.com/illinois-ceesd/mirgecom/pull/875#discussion_r1261414281

Just keeping in the code as a reminder for the coupling PR which will effectively use the models. I suggest we postpone the discussion on this.